media/editor/dataDisplayContext.tsx

```diff
@@ -74,6 +74,19 @@ export class DisplayContext {
 		});
 	}
 
+	/**
+	 * Emitter that fires when the selection changes.
+	 */
+	public onDidChangeAnySelection(listener: (selection: readonly Range[]) => void): IDisposable {
+		messageHandler.sendEvent({
+			type: MessageType.SetSelectedCount,
+			count: this.countSelections(),
+		});
```

Comment:
This function is only called once to install the listener, which is why the bytes aren't syncing. Instead you'd want to do something like

```
this.selectionChangeEmitter.addListener(() => {
		messageHandler.sendEvent({
			type: MessageType.SetSelectedCount,
			count: this.countSelections(),
		});
})
```

within the constructor

Reply:
ah, thanks for catching that! i think i pushed this before i has a chance to fully test all this